lib/_http_agent.js

```diff
@@ -167,6 +172,9 @@ Agent.prototype.addRequest = function addRequest(req, options, port/* legacy */,
     var socket = this.freeSockets[name].shift();
     // Guard against an uninitialized or user supplied Socket.
     if (socket._handle && typeof socket._handle.asyncReset === 'function') {
+      if (destroyHooksExist() && socket._handle.getAsyncId()) {
+        emitDestroy(socket._handle.getAsyncId());
```

Comment:
If we do the destroy from JS, can we cache the result of `getAsyncId()`? Calls into C++ are, unfortunately, not the cheapest…

Reply:
That call no longer exists in #23272.